plugins/module_utils/network/sonic/argspec/ntp/ntp.py

```diff
@@ -0,0 +1,89 @@
+#
+# -*- coding: utf-8 -*-
+# Copyright 2022 Dell Inc. or its subsidiaries. All Rights Reserved
+# GNU General Public License v3.0+
+# (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+#############################################
+#                WARNING                    #
+#############################################
+#
+# This file is auto generated by the resource
+#   module builder playbook.
+#
+# Do not edit this file manually.
+#
+# Changes to this file will be over written
+#   by the resource module builder.
+#
+# Changes should be made in the model used to
+#   generate this file or in the resource module
+#   builder template.
+#
+#############################################
+
+"""
+The arg spec for the sonic_ntp module
+"""
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+class NtpArgs(object):  # pylint: disable=R0903
+    """The arg spec for the sonic_ntp module
+    """
+
+    def __init__(self, **kwargs):
+        pass
+
+    argument_spec = {
+        'config': {
+            'options': {
+                'enable_ntp_auth': {'type': 'bool'},
+                'ntp_keys': {
+                    'elements': 'dict',
+                    'options': {
+                        'encrypted': {'type': 'bool'},
+                        'key_id': {'required': True,
+                                   'type': 'int',
+                                   'no_log': True},
+                        'key_type': {'type': 'str',
+                                     'choices': ['NTP_AUTH_SHA1',
+                                                 'NTP_AUTH_MD5',
+                                                 'NTP_AUTH_SHA2_256']},
+                        'key_value': {'type': 'str', 'no_log': True}
+                    },
+                    'type': 'list',
+                    'no_log': True
```

Comment:
Please remove the "no_log" attribute at the top level for "ntp_keys". It prevents logging of the values that are not "secret". The "key_value" attribute is already protected in this argspec with the "no_log" attribute, so that should be sufficient for providing the needed protection.

Reply:
If remove "no_log" for "ntp_keys", , ther is sanity error:
ERROR: plugins/modules/sonic_ntp.py:0:0: no-log-needed: Argument 'ntp_keys' in argument_spec could be a secret, though doesn't have `no_log` set found in config

Reply:
Thanks for explaining.